ipywidgets/widgets/widget_selectioncontainer.py

```diff
@@ -80,3 +86,13 @@ class Tab(_SelectionContainer):
     """Displays children each on a separate accordion tab."""
     _view_name = Unicode('TabView').tag(sync=True)
     _model_name = Unicode('TabModel').tag(sync=True)
+
+    def __init__(self, **kwargs):
+        if 'children' in kwargs and 'selected_index' not in kwargs and len(kwargs['children']) > 0:
+            kwargs['selected_index'] = 0
+        super(Tab, self).__init__(**kwargs)
+
+    @observe('children')
+    def _observe_children(self, change):
+        if len(change.new) > 0:
+            self.selected_index = 0
```

Comment:
This means that if I keep adding a tab to the end (I think @rmenegaux did this, for example), the selected index keeps being reset to 0.

Reply:
That is right. Although, we don't have operational transforms on sequences, we change all at once.

Since setting the selected index to `None` is not something we want for tabs, this is one way to go.